### PR TITLE
Add parameters for basic handling of mods

### PR DESCRIPTION
**Pull request by sdsznsk**

### CODE REVIEW

**Comment by sdsznsk** [comment]
Also, one question to which I probably know an answer but want to ask first- do we want safeguards (like check if both mutually exclusive are defined)? I saw that there are next to no checks for params so...

**Comment by veteran29** [comment]
LGTM but I don't know Python so leaving the approval to someone else 😁

**Comment by BrettMayson** [comment]
Could allow JSON and the IDs in the env, load the json first then append from env

**Comment by sdsznsk** [comment]
Sounds good, will do

**Comment by Dahlgren** [comment]
Name is only for human readability and not needed by the server.

I suggest we only use one environment variable to specify mods, `GAME_MODS`, which accepts either any version of specific version.

1. Mods with any versions, `GAME_MODS=5965770215E93269,59664C0CB36501CD`
2. Mods with specific versions, `GAME_MODS=5965770215E93269=1.0.6,59664C0CB36501CD=0.3.0`
3. Mods with any and specific versions, `GAME_MODS=5965770215E93269,59664C0CB36501CD=0.3.0`

**Comment by Dahlgren** [comment]
Workshop should use a volume to avoid redownload when container is recreated, for example `VOLUME /reforger/workshop`

To use the volume and separate workshop folder start the game with argument `-addonDownloadDir /reforger/workshop`

**Comment by sdsznsk** [comment]
While I do agree with an option for specifying mod version in the GAME_MODS_IDS_LIST, and keeping them on a dedicated volume, I don't think removing an option to provide mods via a file is a good idea tbh. Managing dozens of mods only via one long ENV would be quite a pain. Your call tho

File: launch.py [hunk 1]
```diff
@@ -109,6 +111,31 @@ def bool_str(text):
...
+    if env_defined("GAME_MODS_IDS_LIST"):
+        reg = re.compile("^[A-Z0-9,=.]+$")
+        if not reg.match(str(os.environ["GAME_MODS_IDS_LIST"])):
+            print("Illegal characters in GAME_MODS_IDS_LIST env")
+            sys.exit(1)
+        mods = str(os.environ["GAME_MODS_IDS_LIST"]).split(",")
+        mods[:] = [mod for mod in mods if mod]  # Remove empty items form list
+        for mod in mods:
+            mod_details = mod.split("=")
+            assert 0 < len(mod_details) < 3, f"{mod} mod not defined properly"
```
[the remark this answers is not in the document]
Comment (sdsznsk) [reply launch.py:122]: Converted all to `assert` statements

**Review by BrettMayson — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -26,10 +27,40 @@ Simply check-out / copy [the provided docker-compose.yml](docker-compose.yml) an
...
+
+a comma separated list of IDs, with an optional version, for example
+
+```sh
+-e GAME_MODS_IDS_LIST="5965770215E93269=1.0.6,5965550F24A0C152"
+```
+
+#### GAME_MODS_JSON_FILE_PATH
+
+path to a JSON file that contains array of mod objects, for example
```
Comment (jonpas) [inline README.md:52]:
```suggestion
Path to a JSON file that contains array of mod objects.
```

File: README.md [hunk 3]
```diff
@@ -26,10 +27,40 @@ Simply check-out / copy [the provided docker-compose.yml](docker-compose.yml) an
...
 
 Alternatively, change the `ARMA_CONFIG` variable to a file present in the `Configs` volume. It will be used without modification.
+
+### Mods
+
+Workshop mods can be defined in two ways. You can use both or either of those.
+
+#### GAME_MODS_IDS_LIST
+
+a comma separated list of IDs, with an optional version, for example
```
Comment (jonpas) [inline README.md:44]:
```suggestion
A comma separated list of IDs, with an optional version.
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: launch.py [hunk 4]
```diff
@@ -109,6 +110,29 @@ def bool_str(text):
...
+                assert reg.match(
+                    mod_details[1]
+                ), f"{mod} mod version does not match the pattern"
+                mod_config["version"] = mod_details[1]
+            config["game"]["mods"].append(mod_config)
+    if env_defined("GAME_MODS_JSON_FILE_PATH"):
+        with open(os.environ["GAME_MODS_JSON_FILE_PATH"]) as f:
+            json_mods = json.load(f)
+            for mod in json_mods:
+                config["game"]["mods"].append(mod)
```
[the remark this answers is not in the document]
Comment (sdsznsk) [reply launch.py:141]: Added in https://github.com/acemod/docker-reforger/pull/11/commits/2a059d5bd40e5a82c724502893d0e7590ccd015d

**Review by jonpas — APPROVED** [review]

**Comment by jonpas** [comment]
@Dahlgren re-review please.

**Review by Dahlgren — APPROVED** [review]
LGTM!

**Comment by sdsznsk** [comment]
Lint failed though, I will take a look in an hr or two 👀

**Pull request merged**